t/unit-tests/u-example-decorate.c

```diff
@@ -11,64 +11,84 @@ struct test_vars {
 	int decoration_a, decoration_b;
 };
 
+static struct test_vars vars;
+
 static void t_add(struct test_vars *vars)
 {
 	void *ret = add_decoration(&vars->n, vars->one, &vars->decoration_a);
 
-	check(ret == NULL);
+	cl_assert(ret == NULL);
 	ret = add_decoration(&vars->n, vars->two, NULL);
-	check(ret == NULL);
+	cl_assert(ret == NULL);
 }
 
 static void t_readd(struct test_vars *vars)
 {
-	void *ret = add_decoration(&vars->n, vars->one, NULL);
+	void *ret = add_decoration(&vars->n, vars->one, &vars->decoration_a);
+	ret = add_decoration(&vars->n, vars->one, NULL);
 
-	check(ret == &vars->decoration_a);
+	cl_assert(ret == &vars->decoration_a);
 	ret = add_decoration(&vars->n, vars->two, &vars->decoration_b);
-	check(ret == NULL);
+	cl_assert(ret == NULL);
 }
 
 static void t_lookup(struct test_vars *vars)
 {
-	void *ret = lookup_decoration(&vars->n, vars->one);
+	void *ret;
 
-	check(ret == NULL);
+	add_decoration(&vars->n, vars->two, &vars->decoration_b);
```

Comment:
This looks mostly as expected, but you'd also have to `add_decoration(&vars->n, vars->one, NULL)` to remain true to the original. Again, this asserts that we can look up `NULL` values.